lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +300,76 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
     end
   end
 
+  def safe_partman_create_parent(table, key:, interval:, **options)
+    raise ArgumentError, "Expected <key> to be present" unless key.present?
+    raise ArgumentError, "Expected <interval> to be present" unless interval.present?
+
+    invalid_options = options.keys - PARTMAN_CREATE_PARENT_OPTIONS
+
+    raise ArgumentError, "Unrecognized optional argument(s): #{invalid_options}" unless invalid_options.empty?
+
+    if ActiveRecord::Base.connection.postgresql_version < 10_00_00
+      raise PgHaMigrations::InvalidMigrationError, "Native partitioning not supported on Postgres databases before version 10"
```

Comment:
I think we should explicitly deprecate (i.e., remove) < PG11 sooner rather than later.

Reply:
I'm not sure it makes sense to completely disallow < PG11 for the entire library, but maybe it makes sense for new features (especially this feature since partman recommends PG11+)

Reply:
> I'm not sure it makes sense to completely disallow < PG11 for the entire library

I'm sure :D

PG11 itself will be EOL this fall; anything else has already been EOL for more than half a year, and that after a 5 year cycle. I think it's completely natural to follow the PostgreSQL projections EOL policy.

Reply:
Fair enough, but let's tackle that in another PR. In the meantime I pushed up a commit to change this guard from < 10 to < 11